File: C_Programming/Practise_Programs/PP_034.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_034
//
//  Description       : This program accepts a number from the user
//                      and displays numbers from 0 to the given
//                      number using a for loop.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 08/07/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include <stdio.h>

//////////////////////////////////////////////////////////////////
//
//  Function Name     : Display()
//
//  Description       : It is used to display numbers from 0 to
//                      the given number using a for loop.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 08/07/2026
//
//////////////////////////////////////////////////////////////////

void Display(int iNo)
{
    int iCnt = 0;

    for(iCnt = 0; iCnt <= iNo; iCnt++)
    {
        printf("%d\t", iCnt);
    }

    printf("\n");
}

int main(void)
{
    int iValue = 0;

    printf("Enter number : ");
    scanf("%d", &iValue);

    Display(iValue);

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter number : 5
//
//  Sample Output :
//      0    1    2    3    4    5
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_034.java
/*
==============================================================================
 Program     : Calculate Sum of Digits

 Description : Reads an integer from the user and calculates the sum of all
               digits present in the specified number.

 Author      : Shubham Gadhe
==============================================================================
*/

import java.util.Scanner;

/*
==============================================================================
 Class       : DigitX

 Description : Provides utility methods for performing operations on digits.
==============================================================================
*/
class DigitX
{
    /*
    ==============================================================================
     Function    : SumDigits

     Description : Calculates and returns the sum of all digits present in the
                   specified number.

     Parameters  : iNo - Number whose digits are to be processed.

     Returns     : Sum of all digits present in the specified number.

     Complexity  : Time  : O(d)
                   Space : O(1)

     Note        : d represents the number of digits in the given number.
    ==============================================================================
    */
    public int SumDigits(int iNo)
    {
        int iSum = 0;

         while(iNo != 0)
        {
            iSum = iSum + (iNo % 10);             
            iNo = iNo / 10;      
        }  

        return iSum;
    }
}

class PP_034
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        DigitX dobj = new DigitX();

        int iValue = 0;
        int iRet = 0;
        
        System.out.println("Enter Number : ");
        iValue = sobj.nextInt();

        iRet = dobj.SumDigits(iValue);    

        System.out.println("Summation of Digits: " + iRet);
    }
}

/*
==============================================================================
 Sample Input
------------------------------------------------------------------------------
 751

 Sample Output
------------------------------------------------------------------------------
 Summation of Digits : 13
==============================================================================
*/